Cue/app/common/TextEntryTableRow.js

```diff
@@ -0,0 +1,120 @@
+// @flow
+
+import React from 'react'
+import { View, Text, TextInput, Platform } from 'react-native'
+
+import CueColors from './CueColors'
+import TableRow from './TableRow'
+
+const styles = {
+  textInput: {
+    width: '100%',
+    marginTop: -4,
+    padding: 0,
+    color: CueColors.primaryText,
+    fontSize: Platform.OS === 'android' ? 14 : 17,
+  },
+  underline: {
+    width: '100%',
+    height: 2,
+  },
+  underlineUnfocused: {
+    backgroundColor: CueColors.lightGrey,
+  },
+  underlineFocused: {
+    backgroundColor: CueColors.primaryTint,
+  }
+}
+
+type Props = {
+  initialText?: string,
+  maxLength?: number,
+  placeholder?: string,
+  onTextChange?: (string) => void
+}
+
+export default class TextEntryTableRow extends React.Component {
+  state: {
+    text: string,
+    height: number,
+    focused: boolean,
+  }
+
+  constructor(props: Props) {
+    super(props)
+
+    this.state = {
+      text: this.props.initialText || '',
+      height: 0,
+      focused: false,
+    }
+  }
+
+  // We need both onChange and onContentSizeChange for height since:
+  //  - onContentSizeChange only fires on initial mount on Android, instead of
+  //    whenever the content size changes (as expected, like on iOS)
+  //  - onChange doesn't fire on initial mount so we can't rely on it to set
+  //    the initial size of the text box
+  _onChange = ({nativeEvent: {contentSize: {height}, text}}) => {
+    this.setState({
+      text,
```

Comment:
It's a [destructuring assignment](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Pulling_fields_from_objects_passed_as_function_parameter) for extracting items from a function parameter. Functionally it's equivalent to:
```js
_onChange = (event) => {
  let height = event.nativeEvent.contentSize.height
  let text = event.nativeEvent.text
  // ...
}